### ISSUE TITLE
The `lite` feature doesn't actually disable `regex`

### ISSUE THREAD

**Issue by msrd0** [Canop/lazy-regex#25]
This is because the `lazy-regex-proc_macros` crate still depends on `regex` instead of `regex-lite` when enabling the `lite` feature.

**Comment by Canop** [comment]
I don't think it matters: the complete regex lib is used for compilation but only regex-lite is used in your compiled application.

You can check for example that the binary is much smaller when you use the "lite" feature.

**Comment by msrd0** [comment]
Well, it might not get included in the final binary, but it definitely increases compile time.

**Comment by Canop** [comment]
Hum, probably.

The reason I didn't propagate the features into the proc macros is that it complexifies notably its code, for example for the handling of the Bytes variant.

I'll see if I can introduce this without making it too messy.
